lib/url.ts

```diff
@@ -1,8 +1,56 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-export interface URLQuery {
-  [queryParameterName: string]: string;
+export class URLQuery {
```

Comment:
jsdoc comments for the class, properties and methods are missing. It is a good idea to document them when everything is fresh in your mind.

Reply:
Added.